web/source/text/engineDeviceSpec.ts

```diff
@@ -0,0 +1,76 @@
+namespace com.keyman.text {
+  export enum Browser {
+    Chrome = 'chrome',
+    Edge = 'edge',
+    Firefox = 'firefox',
+    Native = 'native', // Used by embedded mode
+    Opera = 'opera',
+    Safari = 'safari',
+    Other = 'other'
+  }
+
+  export enum OperatingSystem {
```

Comment:
Do these strings have specific meaning used elsewhere? If not, I'd suggest we make them lower case as well, and use `'macos'` instead of `'MacOSX'`.

Reply:
The issue is that kmwdevice.ts capitalizes these, so I'd need to change the capitalization scheme there.

The capitalization scheme for all of this directly corresponds to the current state of kmwdevice.ts.  Not that we can't change it, of course.

Reply:
I see. So long as nothing else outside KeymanWeb (including other people's code) ever referred to these constants...

Reply:
So I was thinking these were new constants. If they are existing constants, then it may not be safe to change them at all. We may need to maintain the existing strings, case and all.

Reply:
With the commits I added later, these locations are separated.  kmwdevice.ts continues to use the same old constants, while `EngineDeviceSpec` uses the new variants, which are always lower-cased.

Reply:
Okay, thanks. It's kinda a shame that we need to use the old style `macosx` rather than `macos` or even `mac`. But it probably doesn't matter.